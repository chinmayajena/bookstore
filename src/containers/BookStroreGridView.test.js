import React from "react";
import { shallow, mount } from "enzyme";
import ConnectedBookStoreGridView, {
  BookStoreGridView
} from "./BookStoreGridView.js";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";

let initialState = [
  {
    id: 1,
    name: "The Alchemist",
    author: "Paulo Cohelo",
    description: "A philosophical book regarding life journey",
    category: "fiction",
    content: "Some book content",
    isBookmarked: false
  },
  {
    id: 2,
    name: "The Digital Fortress",
    author: "Dawn Brown",
    description: "A techno-thriller novel ",
    category: "fiction",
    content: "Some book content",
    isBookmarked: false
  },
  {
    id: 3,
    name: "Harry Potter",
    author: "J K Rowling",
    description: "A philosophical book regarding life journey",
    category: "fiction",
    content: "Some book content",
    isBookmarked: true
  },
  {
    id: 4,
    name: "Kane n Abel",
    author: "Jeffry Archer",
    description: "A philosophical book regarding life journey",
    category: "fiction",
    content: "Some book content",
    isBookmarked: false
  },

  {
    id: 5,
    name: "Start With Why",
    author: "Simon Sinek",
    description: "A philosophical book regarding life journey",
    category: "non-fiction",
    content: "Some book content",
    isBookmarked: false
  }
];
const mockDeletefn = jest.fn();
const mockBookmarkfn = jest.fn();
const mockUpdateProgressfn = jest.fn();
let appHistory = [];

describe("<BookStoreFilter />", () => {
  let component;

  beforeEach(() => {
    component = shallow(
      <BookStoreGridView
        books={initialState}
        deleteBook={mockDeletefn}
        bookmarkBook={mockBookmarkfn}
        updateBookProgress={mockUpdateProgressfn}
        history={appHistory}
      />
    );
  });

  it("renders the component", () => {
    expect(component.debug()).toMatchSnapshot();
    const wrapper = component.find("ErrorBoundary");
    expect(wrapper.length).toBe(1);
  });

  it("deleteBook is called when dropdwon value changes", () => {
    component.find("#delete-btn-gridview-1").simulate("click");
    expect(mockDeletefn.mock.calls.length).toBe(1);
  });

  it("bookmarkBook is called when input value changes", () => {
    component.find("#bookmark-btn-gridview-1").simulate("click");
    expect(mockBookmarkfn.mock.calls.length).toBe(1);
  });

  it("updateBookProgress is called when button clicked", () => {
    component.find("#edit-btn-gridview-1").simulate("click");
    expect(mockUpdateProgressfn.mock.calls.length).toBe(1);
  });
});

describe("<ConnectedBookStoreFilter />", () => {
  const mockStore = configureStore();
  let store, container;

  beforeEach(() => {
    store = mockStore(initialState);
    store.clearActions();

    container = mount(
      <Provider store={store}>
        <BookStoreGridView
          books={initialState}
          deleteBook={mockDeletefn}
          bookmarkBook={mockBookmarkfn}
          updateBookProgress={mockUpdateProgressfn}
        />
      </Provider>
    );
  });

  it("+++ render the connected component", () => {
    expect(container.debug()).toMatchSnapshot();
    expect(container.length).toEqual(1);
  });
});
