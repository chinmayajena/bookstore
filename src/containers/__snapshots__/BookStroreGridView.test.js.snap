// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookStoreGridView /> renders the component 1`] = `
"<ErrorBoundary>
  <div>
    <div className=\\"row p6 mt4\\">
      <div className=\\"col-md-3\\">
         
        <div className=\\"mt-3 pt-5 card shadow\\">
          <div className=\\"card-body\\">
            <h4 className=\\"card-title\\">
              The Alchemist
            </h4>
            <p className=\\"card-text\\">
              <b>
                Author:
              </b>
              Paulo Cohelo
            </p>
            <p className=\\"card-text\\">
              <b>
                Category:
              </b>
              fiction
            </p>
            <p className=\\"card-text\\">
              <b>
                Description:
              </b>
              A philosophical book regarding life journey
            </p>
            <div className=\\"card-footer\\">
              <button id=\\"edit-btn-gridview-1\\" onClick={[Function: onClick]} className=\\"btn btn-success btn-sm\\">
                <i className=\\"fa fa-edit\\" aria-hidden=\\"true\\" />
              </button>
              <button id=\\"bookmark-btn-gridview-1\\" onClick={[Function: onClick]} className=\\"btn btn-warning btn-sm\\">
                <i className=\\"far fa-star\\" />
              </button>
              <button id=\\"delete-btn-gridview-1\\" onClick={[Function: onClick]} className=\\"btn btn-danger float-right btn-sm\\">
                <i className=\\"fa fa-trash\\" aria-hidden=\\"true\\" />
              </button>
               
            </div>
          </div>
        </div>
         
      </div>
      <div className=\\"col-md-3\\">
         
        <div className=\\"mt-3 pt-5 card shadow\\">
          <div className=\\"card-body\\">
            <h4 className=\\"card-title\\">
              The Digital Fortress
            </h4>
            <p className=\\"card-text\\">
              <b>
                Author:
              </b>
              Dawn Brown
            </p>
            <p className=\\"card-text\\">
              <b>
                Category:
              </b>
              fiction
            </p>
            <p className=\\"card-text\\">
              <b>
                Description:
              </b>
              A techno-thriller novel 
            </p>
            <div className=\\"card-footer\\">
              <button id=\\"edit-btn-gridview-2\\" onClick={[Function: onClick]} className=\\"btn btn-success btn-sm\\">
                <i className=\\"fa fa-edit\\" aria-hidden=\\"true\\" />
              </button>
              <button id=\\"bookmark-btn-gridview-2\\" onClick={[Function: onClick]} className=\\"btn btn-warning btn-sm\\">
                <i className=\\"far fa-star\\" />
              </button>
              <button id=\\"delete-btn-gridview-2\\" onClick={[Function: onClick]} className=\\"btn btn-danger float-right btn-sm\\">
                <i className=\\"fa fa-trash\\" aria-hidden=\\"true\\" />
              </button>
               
            </div>
          </div>
        </div>
         
      </div>
      <div className=\\"col-md-3\\">
         
        <div className=\\"mt-3 pt-5 card shadow\\">
          <div className=\\"card-body\\">
            <h4 className=\\"card-title\\">
              Harry Potter
            </h4>
            <p className=\\"card-text\\">
              <b>
                Author:
              </b>
              J K Rowling
            </p>
            <p className=\\"card-text\\">
              <b>
                Category:
              </b>
              fiction
            </p>
            <p className=\\"card-text\\">
              <b>
                Description:
              </b>
              A philosophical book regarding life journey
            </p>
            <div className=\\"card-footer\\">
              <button id=\\"edit-btn-gridview-3\\" onClick={[Function: onClick]} className=\\"btn btn-success btn-sm\\">
                <i className=\\"fa fa-edit\\" aria-hidden=\\"true\\" />
              </button>
              <button id=\\"bookmark-btn-gridview-3\\" onClick={[Function: onClick]} className=\\"btn btn-warning btn-sm\\">
                <i className=\\"fas fa-star\\" />
              </button>
              <button id=\\"delete-btn-gridview-3\\" onClick={[Function: onClick]} className=\\"btn btn-danger float-right btn-sm\\">
                <i className=\\"fa fa-trash\\" aria-hidden=\\"true\\" />
              </button>
               
            </div>
          </div>
        </div>
         
      </div>
      <div className=\\"col-md-3\\">
         
        <div className=\\"mt-3 pt-5 card shadow\\">
          <div className=\\"card-body\\">
            <h4 className=\\"card-title\\">
              Kane n Abel
            </h4>
            <p className=\\"card-text\\">
              <b>
                Author:
              </b>
              Jeffry Archer
            </p>
            <p className=\\"card-text\\">
              <b>
                Category:
              </b>
              fiction
            </p>
            <p className=\\"card-text\\">
              <b>
                Description:
              </b>
              A philosophical book regarding life journey
            </p>
            <div className=\\"card-footer\\">
              <button id=\\"edit-btn-gridview-4\\" onClick={[Function: onClick]} className=\\"btn btn-success btn-sm\\">
                <i className=\\"fa fa-edit\\" aria-hidden=\\"true\\" />
              </button>
              <button id=\\"bookmark-btn-gridview-4\\" onClick={[Function: onClick]} className=\\"btn btn-warning btn-sm\\">
                <i className=\\"far fa-star\\" />
              </button>
              <button id=\\"delete-btn-gridview-4\\" onClick={[Function: onClick]} className=\\"btn btn-danger float-right btn-sm\\">
                <i className=\\"fa fa-trash\\" aria-hidden=\\"true\\" />
              </button>
               
            </div>
          </div>
        </div>
         
      </div>
      <div className=\\"col-md-3\\">
         
        <div className=\\"mt-3 pt-5 card shadow\\">
          <div className=\\"card-body\\">
            <h4 className=\\"card-title\\">
              Start With Why
            </h4>
            <p className=\\"card-text\\">
              <b>
                Author:
              </b>
              Simon Sinek
            </p>
            <p className=\\"card-text\\">
              <b>
                Category:
              </b>
              non-fiction
            </p>
            <p className=\\"card-text\\">
              <b>
                Description:
              </b>
              A philosophical book regarding life journey
            </p>
            <div className=\\"card-footer\\">
              <button id=\\"edit-btn-gridview-5\\" onClick={[Function: onClick]} className=\\"btn btn-success btn-sm\\">
                <i className=\\"fa fa-edit\\" aria-hidden=\\"true\\" />
              </button>
              <button id=\\"bookmark-btn-gridview-5\\" onClick={[Function: onClick]} className=\\"btn btn-warning btn-sm\\">
                <i className=\\"far fa-star\\" />
              </button>
              <button id=\\"delete-btn-gridview-5\\" onClick={[Function: onClick]} className=\\"btn btn-danger float-right btn-sm\\">
                <i className=\\"fa fa-trash\\" aria-hidden=\\"true\\" />
              </button>
               
            </div>
          </div>
        </div>
         
      </div>
    </div>
     
  </div>
</ErrorBoundary>"
`;

exports[`<ConnectedBookStoreGridView /> +++ render the connected component 1`] = `
"<Provider store={{...}}>
  <Connect(BookStoreGridView) books={{...}} deleteBook={[Function: mockConstructor]} bookmarkBook={[Function: mockConstructor]} updateBookProgress={[Function: mockConstructor]}>
    <BookStoreGridView books={{...}} deleteBook={[Function]} bookmarkBook={[Function]} updateBookProgress={[Function]}>
      <ErrorBoundary>
        <div>
          <div className=\\"row p6 mt4\\">
            <div className=\\"col-md-3\\">
               
              <div className=\\"mt-3 pt-5 card shadow\\">
                <div className=\\"card-body\\">
                  <h4 className=\\"card-title\\">
                    Harry Potter
                  </h4>
                  <p className=\\"card-text\\">
                    <b>
                      Author:
                    </b>
                    J K Rowling
                  </p>
                  <p className=\\"card-text\\">
                    <b>
                      Category:
                    </b>
                    fiction
                  </p>
                  <p className=\\"card-text\\">
                    <b>
                      Description:
                    </b>
                    A philosophical book regarding life journey
                  </p>
                  <div className=\\"card-footer\\">
                    <button id=\\"edit-btn-gridview-3\\" onClick={[Function: onClick]} className=\\"btn btn-success btn-sm\\">
                      <i className=\\"fa fa-edit\\" aria-hidden=\\"true\\" />
                    </button>
                    <button id=\\"bookmark-btn-gridview-3\\" onClick={[Function: onClick]} className=\\"btn btn-warning btn-sm\\">
                      <i className=\\"fas fa-star\\" />
                    </button>
                    <button id=\\"delete-btn-gridview-3\\" onClick={[Function: onClick]} className=\\"btn btn-danger float-right btn-sm\\">
                      <i className=\\"fa fa-trash\\" aria-hidden=\\"true\\" />
                    </button>
                     
                  </div>
                </div>
              </div>
               
            </div>
            <div className=\\"col-md-3\\">
               
              <div className=\\"mt-3 pt-5 card shadow\\">
                <div className=\\"card-body\\">
                  <h4 className=\\"card-title\\">
                    Kane n Abel
                  </h4>
                  <p className=\\"card-text\\">
                    <b>
                      Author:
                    </b>
                    Jeffry Archer
                  </p>
                  <p className=\\"card-text\\">
                    <b>
                      Category:
                    </b>
                    fiction
                  </p>
                  <p className=\\"card-text\\">
                    <b>
                      Description:
                    </b>
                    A philosophical book regarding life journey
                  </p>
                  <div className=\\"card-footer\\">
                    <button id=\\"edit-btn-gridview-4\\" onClick={[Function: onClick]} className=\\"btn btn-success btn-sm\\">
                      <i className=\\"fa fa-edit\\" aria-hidden=\\"true\\" />
                    </button>
                    <button id=\\"bookmark-btn-gridview-4\\" onClick={[Function: onClick]} className=\\"btn btn-warning btn-sm\\">
                      <i className=\\"far fa-star\\" />
                    </button>
                    <button id=\\"delete-btn-gridview-4\\" onClick={[Function: onClick]} className=\\"btn btn-danger float-right btn-sm\\">
                      <i className=\\"fa fa-trash\\" aria-hidden=\\"true\\" />
                    </button>
                     
                  </div>
                </div>
              </div>
               
            </div>
            <div className=\\"col-md-3\\">
               
              <div className=\\"mt-3 pt-5 card shadow\\">
                <div className=\\"card-body\\">
                  <h4 className=\\"card-title\\">
                    Start With Why
                  </h4>
                  <p className=\\"card-text\\">
                    <b>
                      Author:
                    </b>
                    Simon Sinek
                  </p>
                  <p className=\\"card-text\\">
                    <b>
                      Category:
                    </b>
                    non-fiction
                  </p>
                  <p className=\\"card-text\\">
                    <b>
                      Description:
                    </b>
                    A philosophical book regarding life journey
                  </p>
                  <div className=\\"card-footer\\">
                    <button id=\\"edit-btn-gridview-5\\" onClick={[Function: onClick]} className=\\"btn btn-success btn-sm\\">
                      <i className=\\"fa fa-edit\\" aria-hidden=\\"true\\" />
                    </button>
                    <button id=\\"bookmark-btn-gridview-5\\" onClick={[Function: onClick]} className=\\"btn btn-warning btn-sm\\">
                      <i className=\\"far fa-star\\" />
                    </button>
                    <button id=\\"delete-btn-gridview-5\\" onClick={[Function: onClick]} className=\\"btn btn-danger float-right btn-sm\\">
                      <i className=\\"fa fa-trash\\" aria-hidden=\\"true\\" />
                    </button>
                     
                  </div>
                </div>
              </div>
               
            </div>
            <div className=\\"col-md-3\\">
               
              <div className=\\"mt-3 pt-5 card shadow\\">
                <div className=\\"card-body\\">
                  <h4 className=\\"card-title\\">
                    The Alchemist
                  </h4>
                  <p className=\\"card-text\\">
                    <b>
                      Author:
                    </b>
                    Paulo Cohelo
                  </p>
                  <p className=\\"card-text\\">
                    <b>
                      Category:
                    </b>
                    fiction
                  </p>
                  <p className=\\"card-text\\">
                    <b>
                      Description:
                    </b>
                    A philosophical book regarding life journey
                  </p>
                  <div className=\\"card-footer\\">
                    <button id=\\"edit-btn-gridview-1\\" onClick={[Function: onClick]} className=\\"btn btn-success btn-sm\\">
                      <i className=\\"fa fa-edit\\" aria-hidden=\\"true\\" />
                    </button>
                    <button id=\\"bookmark-btn-gridview-1\\" onClick={[Function: onClick]} className=\\"btn btn-warning btn-sm\\">
                      <i className=\\"far fa-star\\" />
                    </button>
                    <button id=\\"delete-btn-gridview-1\\" onClick={[Function: onClick]} className=\\"btn btn-danger float-right btn-sm\\">
                      <i className=\\"fa fa-trash\\" aria-hidden=\\"true\\" />
                    </button>
                     
                  </div>
                </div>
              </div>
               
            </div>
            <div className=\\"col-md-3\\">
               
              <div className=\\"mt-3 pt-5 card shadow\\">
                <div className=\\"card-body\\">
                  <h4 className=\\"card-title\\">
                    The Digital Fortress
                  </h4>
                  <p className=\\"card-text\\">
                    <b>
                      Author:
                    </b>
                    Dawn Brown
                  </p>
                  <p className=\\"card-text\\">
                    <b>
                      Category:
                    </b>
                    fiction
                  </p>
                  <p className=\\"card-text\\">
                    <b>
                      Description:
                    </b>
                    A techno-thriller novel 
                  </p>
                  <div className=\\"card-footer\\">
                    <button id=\\"edit-btn-gridview-2\\" onClick={[Function: onClick]} className=\\"btn btn-success btn-sm\\">
                      <i className=\\"fa fa-edit\\" aria-hidden=\\"true\\" />
                    </button>
                    <button id=\\"bookmark-btn-gridview-2\\" onClick={[Function: onClick]} className=\\"btn btn-warning btn-sm\\">
                      <i className=\\"far fa-star\\" />
                    </button>
                    <button id=\\"delete-btn-gridview-2\\" onClick={[Function: onClick]} className=\\"btn btn-danger float-right btn-sm\\">
                      <i className=\\"fa fa-trash\\" aria-hidden=\\"true\\" />
                    </button>
                     
                  </div>
                </div>
              </div>
               
            </div>
          </div>
           
        </div>
      </ErrorBoundary>
    </BookStoreGridView>
  </Connect(BookStoreGridView)>
</Provider>"
`;
